experimental/devices/bitbang/spi.go

```diff
@@ -84,6 +87,12 @@ func (s *SPI) Connect(f physic.Frequency, mode spi.Mode, bits int) (spi.Conn, er
 	}
 	s.spiConn.mode = mode
 	s.spiConn.bits = bits
+	s.spiConn.readAfterClockPulse = mode&spi.Mode1 == spi.Mode1
+
+	// Set clock idle polarity, ensuring an idle clock to start
+	s.spiConn.clockIdle = gpio.Level(mode&spi.Mode2 == spi.Mode2)
+	s.spiConn.sck.Out(s.spiConn.clockIdle)
```

Comment:
Please check the error. Also set CS too when NoCS is not set, and MISO and MOSI as In()/Out(Low?)

Reply:
Ah ignore MISO/MOSI/CS, that's done above. Just check the error and it'll be fine.

Reply:
Yep, thanks

Reply:
Actually, I'm not so sure all the pin initialization shouldn't be moved to Connect() anyway, in the realm of expected results.  My assumption would be that if there is a separate Connect(), the initial SPI device creation would not modify the hardware.  The rest of the codebase, however, does not appear to make that distinction.  This is definitely in the nit domain.

Reply:
That's a good point.

This was done this way initially so the failure (aka an invalid pin) would occur earlier at SPI instantiation, because Connect() is called within a device driver, which would be confusing for the users.

Reply:
Yeah, that is totally understandable.  Thanks for the reviewing!